incubator/hnc/pkg/kubectl/tree.go

```diff
@@ -34,15 +36,24 @@ var (
 var treeCmd = &cobra.Command{
 	Use:   "tree",
 	Short: "Displays the hierarchy tree rooted at the given namespace",
-	Args:  cobra.ExactArgs(1),
+	Args:  cobra.MaximumNArgs(1),
 	Run: func(cmd *cobra.Command, args []string) {
-		nnm := args[0]
 		footnotesByMsg = map[string]int{}
 		footnotes = []string{}
-		hier := getHierarchy(nnm)
-		fmt.Println(nameAndFootnotes(hier))
-		printSubtree("", hier)
-
+		nsList := make([]string, 0)
+		if len(args) != 0 {
```

Comment:
You use the test `len(args) == 0` several times; you should probably factor this out into a variable to make it more obvious what you're doing. For example:

```
defaultList := len(args) == 0
if defaultList {
  ...
}
```